rs/bitcoin/adapter/src/router.rs

```diff
@@ -50,67 +50,69 @@ pub fn start_main_event_loop(
     tokio::task::spawn(async move {
         let mut tick_interval = interval(Duration::from_millis(100));
         loop {
-            let sleep_idle_interval = Duration::from_millis(100);
-            if adapter_state.is_idle() {
-                connection_manager.make_idle();
-                blockchain_manager.make_idle();
-                // TODO: instead of sleeping here add some async synchronization.
-                sleep(sleep_idle_interval).await;
-                continue;
-            }
+            connection_manager.make_idle();
+            blockchain_manager.make_idle();
```

Comment:
nit: I would move the make_idle at the bottom of this loop

Reply:
Done, makes sense. we don't want to make those idle for nothing, even if we are awake